lib/datadog/appsec/contrib/graphql/gateway/watcher.rb

```diff
@@ -38,11 +38,9 @@ def watch_multiplex(gateway = Instrumentation.gateway)
                           actions: result.actions
                         }
 
-                        if scope.service_entry_span
-                          scope.service_entry_span.set_tag('appsec.blocked', 'true') if result.actions.include?('block')
-                          scope.service_entry_span.set_tag('appsec.event', 'true')
-                        end
-
+                        # We want to keep the trace in case of security event
+                        scope.trace.keep! if scope.trace
+                        Datadog::AppSec::Event.add_tags(scope, result)
```

Comment:
Does it make sense to always bundle these two lines together?

Reply:
If we create a new method to bundle these two line together, I think that `add_tags_and_keep!` or `tag_and_keep!` would be a better name than just `tag!` as `keep!` does more than just setting a tag (see this [review](https://github.com/DataDog/dd-trace-rb/pull/3965#discussion_r1791328487))

Reply:
Fixed in 73f8af2cd11e7e37a9d91a6b0cb1e72e1edfae02